presto-kudu/src/main/java/io/prestosql/plugin/kudu/KuduClientSession.java

```diff
@@ -157,6 +158,11 @@ public Schema getTableSchema(KuduTableHandle tableHandle)
             return ImmutableList.of();
         }
         addConstraintPredicates(table, builder, constraint);
+        TupleDomain<ColumnHandle> dynamicConstraint = dynamicFilter.getCurrentPredicate();
```

Comment:
Why not just intersect this with `tableHandle.getConstraint()` above ?

Reply:
Modified